src/fidesops/api/v1/endpoints/user_endpoints.py

```diff
@@ -17,16 +18,29 @@
 from fidesops.util.oauth_util import verify_oauth_client
 from sqlalchemy.orm import Session
 
-from fidesops.api.v1.scope_registry import (
-    USER_CREATE,
-    USER_DELETE,
-    SCOPE_REGISTRY,
-)
+from fidesops.api.v1.scope_registry import USER_CREATE, USER_DELETE, SCOPE_REGISTRY
 
 logger = logging.getLogger(__name__)
 router = APIRouter(tags=["Users"], prefix=V1_URL_PREFIX)
 
 
+def perform_login(db: Session, user: FidesopsUser) -> ClientDetail:
+    """Performs a login by updating the FidesopsUser instance and
+    creating and returning an associated ClientDetail."""
+
+    client: ClientDetail = user.client
```

Comment:
Hey @adamsachs: are you 100% sure that a logout purges this relation alongside deleting the record in the `ClientDetail` table? It'd be good to have certainty on that part, otherwise this check may return false negatives if say, `user.client` is derived from `user.client_id`, and that's pointing to a deleted `ClientDetail` record.

Reply:
good question, i was curious about the lifecycle here too. to be clear - i don't think my changes should have had any impact on this behavior (though i could be wrong about that). either way, i can see why it's worth confirming the behavior now that our attention is on it.

Reply:
Agree, none of these changes would affect the behaviour, though the behaviour will affect whether this conditional works as intended I believe?

Reply:
right, i guess i'm just saying that the conditional was already in place, and i'd tried not to touch more than the scope of the ticket - but i can see that it's worth validating the behavior here.

a little testing locally seems to confirm your suspicion (i'm getting a failed assertion when i check whether the user's `client` is `None` after the logout is performed in a unit test).

i see a few ways of potentially addressing:
-have the logout endpoint remove the user -> client reference
-upon login (i.e. in this spot), perform a lookup on the referenced client to check if its stale/deleted. if so, create a new client
-just create a new client/session any time the `/login` endpoint is invoked, whether or not the user has logged out. i'm not exactly sure on the intended/expected semantics of `/login`, i.e. if it's necessary (or desirable) that it reuses an existing and valid session

Reply:
Are you sure @adamsachs? (Can you specify what your failed assertion is)  If in testing, **you reload the user object**, I'm fairly sure that the `client` won't exist.  This is only a relationship in the ORM side.

Reply:
@pattisdr helped me out and made me realize i was misinterpreting my local test results. in https://github.com/ethyca/fidesops/pull/394/commits/42740520d8a59a3b815518be664f4d48a208a379 and https://github.com/ethyca/fidesops/pull/394/commits/011d810c6e7b1383a39fbf00bf96a7d6e8b97627 i've updated the test cases to confirm that the user -> client pointer is getting cleaned up, and that requests with outdated client credentials are properly rejected with a 403.